mls-rs-uniffi/src/config.rs

```diff
@@ -56,10 +55,19 @@ impl mls_rs_core::group::GroupStateStorage for ClientGroupStorage {
     }
 }
 
+#[cfg(not(any(feature = "openssl")))]
+compile_error!(
+    "The crypto provider must be set by enabling exactly one of the \
+     following Cargo features: [\"openssl\"]."
+);
+
+#[cfg(feature = "openssl")]
+pub type UniFFICryptoProvider = mls_rs_crypto_openssl::OpensslCryptoProvider;
+
```

Comment:
If we create another feature "boringssl", we won't be able to have `UniFFICryptoProvider` defined twice, but features should be additive. Can we achieve additive features by defining `#[cfg(feature = "openssl")] UniFFIConfigOpenssl = ...` and `#[cfg(feature = "boringssl")] UniFFIConfigBoringssl = ...`?

Reply:
> If we create another feature "boringssl", we won't be able to have `UniFFICryptoProvider` defined twice, but features should be additive.

Yes, ideally... I was going to guard against having multiple features enabled by letting the build fail with `compile_fail!` like you saw above. It's not great, but I don't know of a way to make the crypto provider dynamic at runtime.

What we need is

- a non-generic struct which holds a `mls_rs::Client` inside it
- the `mls_rs::Client` value should be able to hold different `CryptoProvider`s, implying that we should have a `CryptoProviderWrapper` type which `impl CryptoProvider` and which holds a `Box<dyn CryptoProvider>`

So I'm looking for a way to do

```rust
struct CryptoProviderWrapper {
    crypto_provider: Box<dyn CryptoProvider>,
}
```

However, this doesn't compile because I need to specify the `CipherSuiteProvider` associated type. I don't know how I can do this in a non-generic way?